tests/integration/fixtures/processing.py

```diff
@@ -355,14 +355,18 @@ def get_metric(self, timeout=None):
         return json.loads(message.value()), message.headers()
 
     def get_metrics(self, timeout=None, max_attempts=100):
+        metrics = []
+
         for _ in range(max_attempts):
             message = self.poll(timeout=timeout)
 
             if message is None:
-                return
+                break
             else:
                 assert message.error() is None
-                yield json.loads(message.value()), message.headers()
+                metrics.append((json.loads(message.value()), message.headers()))
+
+        return metrics
```

Comment:
Alternatively you could have kept the generator as `_get_metrics`, and make `get_metrics` return `list(self._get_metrics(...))`. But this works too 👍

Reply:
I wanted to save some work for the compiler 😆